### PR TITLE
Write error log if updating entity store fails

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

Remove `unwrap()`s in the c8y mapper actor that would crash the actor if updating the store failed.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/extensions/c8y_mapper_ext/src/actor.rs [hunk 1]
```diff
@@ -108,19 +109,25 @@ impl C8yMapperActor {
...
+                    error!("Could not update device registration: {e}");
+                }
             } else {
                 // if device is unregistered register using auto-registration
                 if self.entity_store.get(entity_id).is_none() {
-                    let register_messages = self.auto_register_entity(entity_id);
+                    let register_messages = match self.auto_register_entity(entity_id) {
+                        Ok(register_messages) => register_messages,
+                        Err(e) => {
+                            error!("Could not update device registration: {e}");
```
Comment (PradeepKiruvale) [inline crates/extensions/c8y_mapper_ext/src/actor.rs:126]: I feel it's better to write this error message to the `error` topic. @didier-wenzek can clarify this.

File: crates/extensions/c8y_mapper_ext/src/actor.rs [hunk 2]
```diff
@@ -108,19 +109,25 @@ impl C8yMapperActor {
...
-        // to which entity is the message related
         if let Some(entity_id) = entity_store::entity_mqtt_id(&message.topic) {
-            // if message is registration message:
             if is_entity_register_message(&message) {
-                // update entity store with registration message
-                // TODO: emit error message on Err
-                self.entity_store
+                if let Err(e) = self
+                    .entity_store
                     .update(message.clone().try_into().unwrap())
```
Comment (PradeepKiruvale) [inline crates/extensions/c8y_mapper_ext/src/actor.rs:116]: Need to handle this as well

File: crates/extensions/c8y_mapper_ext/src/actor.rs [hunk 3]
```diff
@@ -165,8 +175,7 @@ impl C8yMapperActor {
             .with_retain();
```
Comment (PradeepKiruvale) [inline crates/extensions/c8y_mapper_ext/src/actor.rs:175]: You can use `Topic::new_unchecked("")` to create the topic, no need to worry about `unwrap`.

File: crates/extensions/c8y_mapper_ext/src/actor.rs (hunk 1 above)
Comment (didier-wenzek) [reply crates/extensions/c8y_mapper_ext/src/actor.rs:126]: To use or not the error topic is a broader topic that needs to be discussed with @reubenmiller .

Thin-edge emits error messages on too many channels. This is puzzling for the users (what streams  do I have to watch?) and the developers (where do I have to log this error?).

Ideally sending errors over MQTT should be done behind the scene by the `error!()` macro.

File: crates/extensions/c8y_mapper_ext/src/actor.rs (hunk 2 above)
Comment (Bravo555) [reply crates/extensions/c8y_mapper_ext/src/actor.rs:116]: In this case, the message was previously checked to be a register message on line 113, so this unwrap never panics. That's not very obvious to see, so the API will be changed a bit to remove this unwrap in a later PR.

File: crates/extensions/c8y_mapper_ext/src/actor.rs (hunk 3 above)
Comment (Bravo555) [reply crates/extensions/c8y_mapper_ext/src/actor.rs:175]: okay, looks like `new_unchecked` saves an extra check, so I will be using it from now on

**Review by PradeepKiruvale — APPROVED** [review]

**Pull request merged**